cmd/imagePushToRegistry.go

```diff
@@ -0,0 +1,200 @@
+}
+
+func newImagePushToRegistryUtils() imagePushToRegistryUtils {
+	utils := imagePushToRegistryUtilsBundle{
+		Command:          &command.Command{},
+		Files:            &piperutils.Files{},
+		dockerImageUtils: &docker.CraneUtilsBundle{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func imagePushToRegistry(config imagePushToRegistryOptions, telemetryData *telemetry.CustomData) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	utils := newImagePushToRegistryUtils()
+
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runImagePushToRegistry(&config, telemetryData, utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runImagePushToRegistry(config *imagePushToRegistryOptions, telemetryData *telemetry.CustomData, utils imagePushToRegistryUtils) error {
+	if config.TargetImage == "" {
+		config.TargetImage = config.SourceImage
+	}
+
+	re := regexp.MustCompile(`^https?://`)
+	sourceRegistry := re.ReplaceAllString(config.SourceRegistryURL, "")
+	targetRegistry := re.ReplaceAllString(config.TargetRegistryURL, "")
+	src := fmt.Sprintf("%s/%s", sourceRegistry, config.SourceImage)
+	dst := fmt.Sprintf("%s/%s", targetRegistry, config.TargetImage)
+
+	err := handleCredentialsForPrivateRegistry(config.DockerConfigJSON, sourceRegistry, config.SourceRegistryUser, config.SourceRegistryPassword, utils)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle credentials for source registry")
+	}
+
+	err = handleCredentialsForPrivateRegistry(config.DockerConfigJSON, targetRegistry, config.TargetRegistryUser, config.TargetRegistryPassword, utils)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle credentials for target registry")
+	}
+
+	if len(config.LocalDockerImagePath) > 0 {
+		if err := pushLocalImageToTargetRegistry(config.LocalDockerImagePath, dst, config.TagLatest, utils); err != nil {
+			return errors.Wrapf(err, "failed to push local image to %q", targetRegistry)
+		}
+		return nil
+	}
+
+	if err := copyImage(src, dst, config.TagLatest, utils); err != nil {
+		return errors.Wrapf(err, "failed to copy image from %q to %q", sourceRegistry, targetRegistry)
+	}
+
+	return nil
+}
+
+func handleCredentialsForPrivateRegistry(dockerConfigJsonPath, registry, username, password string, utils imagePushToRegistryUtils) error {
+	if len(dockerConfigJsonPath) == 0 && (len(registry) == 0 || len(username) == 0 || len(password) == 0) {
+		return nil
+	}
```

Comment:
I'm not sure if it's correct to simply return nil in this case.. If it's not an error, there should at least be a debug message to help developers.